### PR TITLE
Adds an add_settings_error() wrapper and uses it

### PR DESCRIPTION
**Pull request by manooweb**
Fixes partially https://github.com/polylang/polylang-pro/issues/2060 (Polylang parts)

See [#2060](https://github.com/polylang/polylang-pro/issues/2060) for explanations of what this PR propose.
First step required before applying to Polylang Pro.

Finally, this wrapper function name is `pll_add_notice()`.

We will need to take in account the [Polylang Pro special case](https://github.com/polylang/polylang-pro/blob/3.6-beta1/modules/machine-translation/Module_Settings.php#L129-L144).  ➡ In https://github.com/polylang/polylang-pro/pull/2067 we decided to pass the whole data as a `WP_Error()` extra data string. So the string is passed to `add_settings_error()` $type parameter as before.

### CODE REVIEW

**Review by Screenfeed — APPROVED** [review]

File: include/functions.php [hunk 1]
```diff
@@ -201,3 +201,37 @@ function pll_is_plugin_active( string $plugin_name ) {
...
+ * Prepares and registers errors
+ *
+ * Wraps `add_settings_error()` to make its use more consistent.
+ *
+ * @since 3.6
+ *
+ * @param WP_Error $error Error object.
+ * @return void
+ */
+function pll_add_settings_error( WP_Error $error ) {
```
Comment (Chouby) [inline include/functions.php:215]: As we don't use them only for settings, what do you think to rename the function? Maybe `pll_add_error_notice()` ?

File: include/functions.php [hunk 2]
```diff
@@ -201,3 +201,37 @@ function pll_is_plugin_active( string $plugin_name ) {
...
+		return;
+	}
+
+	foreach ( $error->get_error_codes() as $error_code ) {
+		// Extract the "error" type.
+		$data = $error->get_error_data( $error_code );
+		$type = empty( $data ) || ! is_string( $data ) ? 'error' : $data;
+
+		$message = wp_kses(
+			$error->get_error_message( $error_code ),
```
Comment (Screenfeed) [inline include/functions.php:226]: What about this?
```suggestion
			implode( '<br>', $error->get_error_messages( $error_code ) ),
```
This is done [here](https://github.com/polylang/polylang-pro/blob/fa6667c34f6f0afca29027c0c6a12e80b9f4687c/modules/import-export/export/export-strings-action.php#L161-L162) and [here](https://github.com/polylang/polylang-pro/blob/fa6667c34f6f0afca29027c0c6a12e80b9f4687c/modules/import-export/import/import-action.php#L320-L321).

Comment (Screenfeed) [reply include/functions.php]: Yes, that's because I was reviewing this other PR that I suggested this ;)

File: include/functions.php (hunk 1 above)
Comment (manooweb) [reply include/functions.php]: Renamed to `pll_add_notice()` in bee5b32d5d80cfa3233f596a38193a992be6429c

**Pull request merged**